dotCMS/src/main/java/com/dotcms/rendering/velocity/util/VelocityUtil.java

```diff
@@ -0,0 +1,607 @@
+
+	      return (firstLetterUppercase) 
+	              ? StringUtils.camelCaseUpper(variable)
+	              : StringUtils.camelCaseLower(variable);
+		
+	}
+	
+	
+	public static Boolean isNotAllowedVelocityVariableName(String variable){
+		
+
+		String [] notallwdvelvars={"inode","type", "modDate", "owner", "ownerCanRead", "ownerCanWrite", "ownerCanPublish",
+				"modUser", "working", "live", "deleted", "locked","structureInode", "languageId", "permissions",
+				"identifier", "conHost", "conFolder", "Host", "folder"}; 
+		Boolean found=false;
+		for(String notallowed : notallwdvelvars){
+			 if(variable.equalsIgnoreCase(notallowed)){
+				 found=true;
+			 }
+			
+		}
+		return found;
+	}
+
+	/**
+	 * Returns a basic Velocity Context without any toolbox or request
+	 * response, session objects;
+	 * @return
+	 */
+
+	public static Context getBasicContext() {
+		Context context = new VelocityContext();
+		context.put("UtilMethods", new UtilMethods());
+		context.put("PortletURLUtil", new PortletURLUtil());
+		context.put("quote", "\"");
+		context.put("pounds", "##");
+		context.put("return", "\n");
+		context.put("velocityContext", context);
+		context.put("language", "1");
+		context.put("InodeUtils", new InodeUtils());
+		return context;
+	}
+	
+	
+	/**
+	 * Gets creates Velocity context will all the toolbox, user, host, language and request stuff
+	 * inside the map
+	 * @param request
+	 * @param response
+	 * @return
+	 */
+	public static ChainedContext getWebContext(HttpServletRequest request, HttpServletResponse response) {
+		return getWebContext(getBasicContext(), request, response);
+	}
+	
+	
+	public static ChainedContext getWebContext(Context ctx, HttpServletRequest request, HttpServletResponse response) {
+
+        if ( ctx == null ) {
+            ctx = getBasicContext();
+        }
+
+        // http://jira.dotmarketing.net/browse/DOTCMS-2917
+
+		//get the context from the request if possible
+        ChainedContext context;
+        if ( request.getAttribute( com.dotcms.rendering.velocity.Constants.VELOCITY_CONTEXT ) != null && request.getAttribute( com.dotcms.rendering.velocity.Constants.VELOCITY_CONTEXT ) instanceof ChainedContext ) {
+            return (ChainedContext) request.getAttribute( "velocityContext" );
+        } else {
+            RequestWrapper rw = new RequestWrapper( request );
+            if ( request.getAttribute( "User-Agent" ) != null && request.getAttribute( "User-Agent" ).equals( Constants.USER_AGENT_DOTCMS_BROWSER ) ) {
+                rw.setCustomUserAgentHeader( Constants.USER_AGENT_DOTCMS_BROWSER );
+            }
+            context = new ChainedContext( ctx, getEngine(), rw, response, Config.CONTEXT );
+        }
+
+        context.put("context", context);
+		Logger.debug(VelocityUtil.class, "ChainedContext=" + context);
+		/*
+		 * if we have a toolbox manager, get a toolbox from it See
+		 * /WEB-INF/toolbox.xml
+		 */
+		context.setToolbox(getToolboxManager().getToolboxContext(context));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "setToolbox"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "getToolboxContext"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)